support-payment-api/src/main/scala/backend/StripeBackend.scala

```diff
@@ -148,20 +174,25 @@ class StripeBackend(
               EitherT.fromEither(Left(error))
           }
         }
+    }
 
-    recaptchaRequired().flatMap {
+    stripeEnabled(request).flatMap {
       case true =>
-        recaptchaService
-          .verify(request.recaptchaToken)
-          .flatMap { resp =>
-            if (resp.success) createIntent()
-            else EitherT.leftT(StripeApiError.fromString(recaptchaErrorText, publicKey = None))
-          }
-      case false =>
-        createIntent()
+        recaptchaRequired().flatMap {
+          case true =>
+            recaptchaService
+              .verify(request.recaptchaToken)
+              .flatMap { resp =>
+                if (resp.success) createIntent()
+                else EitherT.leftT(StripeApiError.fromString(recaptchaErrorText, publicKey = None))
+              }
+          case false =>
+            createIntent()
+        }
+      case _ =>
+        EitherT.leftT(StripeApiError.fromString(stripeDisabledErrorText, publicKey = None))
     }
```

Comment:
I think creating another function would help here as well eg.
```suggestion
    def checkRecaptcha = recaptchaRequired().flatMap {
      case true =>
        recaptchaService
          .verify(request.recaptchaToken)
          .flatMap { resp =>
            if (resp.success) createIntent()
            else EitherT.leftT(StripeApiError.fromString(recaptchaErrorText, publicKey = None))
          }
      case false =>
        createIntent()
    }

    stripeEnabled(request).flatMap {
      case true =>
        checkRecaptcha
      case false =>
        EitherT.leftT(StripeApiError.fromString(stripeDisabledErrorText, publicKey = None))
    }
```